python/idsse_common/idsse/common/rabbitmq_utils.py

```diff
@@ -200,24 +201,27 @@ class PublisherSync:
     you're done with it using close().
 
     Args:
-        connection (Conn | BlockingConnection): connection parameters to establish a new
-            RabbitMQ connection, or existing RabbitMQ to reuse
-        params (RabbitMqParams): parameters for RabbitMQ exchange and queue on which to publish
+        conn_params (Conn): connection parameters to establish a new
+            RabbitMQ connection
+        rmq_params (RabbitMqParams): parameters for RabbitMQ exchange and queue on which to publish
             messages
     """
     def __init__(
         self,
-        connection: Conn | BlockingConnection,
-        params: RabbitMqParams,
+        conn_params: Conn,
+        rmq_params: RabbitMqParams,
         channel: Channel | None = None,
     ) -> tuple[BlockingConnection, Channel]:
+        # save params
+        self._conn_params = conn_params
+        self._rmq_params = rmq_params
 
         # establish BlockingConnection and declare exchange and queue on Channel
         self._connection, self._channel, self._queue_name = _initialize_connection_and_channel(
-            connection, params, channel,
+            conn_params, rmq_params, channel,
         )
-        self._exchange_name = params.exchange.name
-        self._channel.confirm_delivery() # enable delivery confirmations from RabbitMQ broker
+        self._exchange_name = rmq_params.exchange.name
```

Comment:
We don't need to save the exchange name individually, I had just done that since it was the only RMQ parameter I cared about to call `basic_publish()`. 

You could drop this private variable and just pass `self._rmq_params.exchange.name` to basic_publish, although it's a bit longer.

Reply:
I made the switch and don't save exchange name twice, thanks.